### PR TITLE
Document why our RFC process is public

### PR DESCRIPTION
**Pull request by tahb**
This proposal comes straight from a now [closed issue](https://github.com/dxw/tech-team-rfcs/issues/2) where we discussed the issue with several members of the team.

The issue debates the position of being open by default and that being open makes things better. 

This proposal seeks to document why this repository is currently public.

There were a few concerns raised around the need to sometimes include sensitive information around incidents. I think we can make use of tech forums to discuss sensitive details offline if we need to as a first pass, if that fails we can make it private later. It is much harder to go in the other direction of making a private repo public.

If we do no approve this I propose we need another RFC that documents the decision to make it private.

### CODE REVIEW

File: rfc-018-our-decision-making-process-is-public.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ it a public one will be incredibly difficult to do safely. By starting with
+ a public repository however, we always have the safe and easy option of
+  making private open to us.
+
+## Proposal
+
+We SHOULD NOT make this repository private.
+
+We SHOULD promote sensitive discussions from GitHub into our regular
+ tech-forums. Here we will discuss the issue in a private forum and we MUST
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]: I disagree that tech forums are the place to have those discussions. Part of the point of using RFCs was to make it easier for people who are not always able to attend those forums to be involved in the decision-making process. Can we rephrase this to just refer to taking the discussion to a more private channel without be strict about it being the forum?

File: rfc-018-our-decision-making-process-is-public.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  publish a safe summary of the discussion to allow voting to continue as
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]:
```suggestion
publish a summary of the discussion with as much detail as is appropriate for public consumption to allow voting to continue as
```

File: rfc-018-our-decision-making-process-is-public.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   normal with adequate reference.
+
+We SHOULD have a retrospectives on how the team have found making decisions
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]:
```suggestion
We SHOULD have retrospectives on how the team have found making decisions
```

File: rfc-018-our-decision-making-process-is-public.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ in public.
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]: How frequently? Is there a maximum time that we feel comfortable setting? At least once every 6 months?

File: rfc-018-our-decision-making-process-is-public.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+We MAY wish to make it private later if we retrospectively find that these
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]:
```suggestion
We MAY wish to make it private later if we find that these
```

File: rfc-018-our-decision-making-process-is-public.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ concerns cannot be addressed using retrospectives and tech forums.
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]:
```suggestion
 concerns cannot be addressed while this repository is public.
```

File: rfc-018-our-decision-making-process-is-public.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  future team decide to make it public
+- most of the application code we write lives in open source repositories where
+our work and decision making process can already be viewed publicly
+
+Reasons against:
+
+- parts of our discussion make have to include commercially sensitive
+ information eg. incident reports
+- as the public can read the team may feel too uncomfortable to contribute to
+ discussions making it harder to talk with candour
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]: There is an additional one in that, deciding to make this private at a later date will likely be in response to something happening that we consider to be too sensitive to discuss in public. Therefore, making this repository private is in itself information leaking.

File: rfc-018-our-decision-making-process-is-public.md (hunk 4 above)
Comment (tahb) [reply rfc-018-our-decision-making-process-is-public.md]: I agree we should be thinking about time so that it is actionable however I'm not sure the RFC is responsible for capturing how we implement this. Perhaps a next step should be to schedule one for 6 months time.

File: rfc-018-our-decision-making-process-is-public.md (hunk 7 above)
Comment (tahb) [reply rfc-018-our-decision-making-process-is-public.md]: I'm afraid I don't follow.

File: rfc-018-our-decision-making-process-is-public.md (hunk 1 above)
Comment (Floppy) [reply rfc-018-our-decision-making-process-is-public.md]: Makes sense to me.

File: rfc-018-our-decision-making-process-is-public.md (hunk 7 above)
Comment (erbridge) [reply rfc-018-our-decision-making-process-is-public.md]: This repository is public. If it becomes private (and therefore disappears for other users), someone could infer that we had something happen that meant we needed to make this repository private to discuss it.

Comment (tahb) [reply rfc-018-our-decision-making-process-is-public.md]: Thanks for re-explaining. I can see that making it private in future could leak the information to the public that there was now content we deemed sensitive in it. The negative outcome for this could be that the private repo could then become a target for people to glean information from?

I've added a point to that affect but if I've missed the mark again I'm happy to update it.

**Comment by tahb** [comment]
I'm sorry for dropping the ball on this for so long. The amount of Hound formatting errors was off putting when the linter worked locally. I have updated the branch with the last(?) bit of feedback and Hound's happy.

I've run `npm run lint` locally and there were no issues found so I _hope_ this is good for approve/reject!

File: rfc-018-our-decision-making-process-is-public.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+This repository started as public by default but do we agree that this is right
+in this case?
+
+Reasons for:
+
+- dxw values include being honest and we agree with the
+  [principle of making things open as it makes things better](https://www.gov.uk/guidance/government-design-principles#make-things-open-it-makes-things-better)
+- [the dxw playbook is public](https://github.com/dxw/playbook) and has not
+ caused us concern or prevented us from iterating it
+- others can learn from us and spend more their time making things better for
```
Comment (lawrence-dxw) [inline rfc-018-our-decision-making-process-is-public.md]:
```suggestion
- others can learn from us and spend more of their time making things better for
```

File: rfc-018-our-decision-making-process-is-public.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  users
+- it allows us to share our thinking and rationale on our technical decisions,
+  doing so publicly as well as privately can increase trust and empathy with
+  more people
+- with the potential for the world watching, the bar is raised.
+  being open from the start ensures that there is less risk to us should a
```
Comment (lawrence-dxw) [inline rfc-018-our-decision-making-process-is-public.md]: Should this be a new bullet?

Comment (erbridge) [reply rfc-018-our-decision-making-process-is-public.md]: I _think_ it's a continuation of the previous point. Probably could do with a captial letter.

File: rfc-018-our-decision-making-process-is-public.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+This repository started as public by default but do we agree that this is right
+in this case?
+
+Reasons for:
+
+- dxw values include being honest and we agree with the
+  [principle of making things open as it makes things better](https://www.gov.uk/guidance/government-design-principles#make-things-open-it-makes-things-better)
+- [the dxw playbook is public](https://github.com/dxw/playbook) and has not
+ caused us concern or prevented us from iterating it
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]: Nit pick if you're back editing this file again:

```suggestion
  caused us concern or prevented us from iterating it
```

File: rfc-018-our-decision-making-process-is-public.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  being open from the start ensures that there is less risk to us should a
+  future team decide to make it public
+- most of the application code we write lives in open source repositories where
+our work and decision making process can already be viewed publicly
+
+Reasons against:
+
+- parts of our discussion make have to include commercially sensitive
+ information eg. incident reports
+- as the public can read the team may feel too uncomfortable to contribute to
```
[the remark this answers is not in the document]
Comment (erbridge) [reply rfc-018-our-decision-making-process-is-public.md]: Still relevant @tahb

File: rfc-018-our-decision-making-process-is-public.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+It's difficult to change our mind if we start from a private repository.
+
+Contributors are aware of the context in which they add their comments. If
+ they make comments under the context of a private repository which then
+ becomes public, there is a potential risk for us to make content that was
+ not intended or written in a way appropriate for public consumption.
+ Therefore starting from a private repository and later deciding to make
+ it a public one will be incredibly difficult to do safely. By starting with
+ a public repository however, we always have the safe and easy option of
+  making private open to us.
```
Comment (erbridge) [inline rfc-018-our-decision-making-process-is-public.md]: Indentation on here's a bit odd. Another nitpick if you're back in here. Not a blocker for merging.

File: rfc-018-our-decision-making-process-is-public.md (hunk 1 above)
Comment (erbridge) [reply rfc-018-our-decision-making-process-is-public.md]: Still relevant @tahb.

File: rfc-018-our-decision-making-process-is-public.md (hunk 2 above)
Comment (erbridge) [reply rfc-018-our-decision-making-process-is-public.md]: Still relevant @tahb.
